src/Box.php

```diff
@@ -117,17 +141,115 @@ public function registerStub(string $file): void
         $this->phar->setStub($contents);
     }
 
+    public function registerFileMapping(callable $retrieveBasePath, callable $mapFile): void
+    {
+        $this->retrieveBasePath = $retrieveBasePath;
+        $this->mapFile = $mapFile;
+    }
+
+    /**
+     * @param iterable|SplFileInfo[] $files
+     */
+    public function addFiles(iterable $files): void
+    {
+        $retrieveBasePath = $this->retrieveBasePath;
+        $mapFile = $this->mapFile;
+        $compactors = $this->compactors;
+        $placeholders = $this->placeholders;
+
+        //Debug: the values passed to the $processFile closure seems to be working fine
+        $x0 = \serialize($retrieveBasePath);
+        $x1 = \serialize($mapFile);
+        $x3 = \serialize($placeholders);
+        $x4 = \serialize($compactors);
+
+        $y0 = \unserialize($x0, []);
+        $y1 = \unserialize($x1, []);
+        $y3 = \unserialize($x3, []);
+        $y4 = \unserialize($x4, []);
+
+        $files = array_map(
+            function (\SplFileInfo $fileInfo): string {
+                return $fileInfo->getPathname();
+            },
+            iterator_to_array($files)
```

Comment:
$files may be an array here.

Reply:
true, it's not in practice but I should account for that still 👍